tests/test_transformation/test_yeojohnson_transformer.py

```diff
@@ -84,3 +85,68 @@ def test_non_fitted_error(df_vartypes):
     with pytest.raises(NotFittedError):
         transformer = YeoJohnsonTransformer()
         transformer.transform(df_vartypes)
+
+
+def test_inverse_transform_automatically_select_only_transformed_columns(df_vartypes):
+    X = df_vartypes.copy(deep=True)
+    transformer = YeoJohnsonTransformer(variables=None)
+    X_trans = transformer.fit_transform(X)
+
+    X_inverse = transformer.inverse_transform(X_trans)
+
+    pd.testing.assert_frame_equal(X, X_inverse, check_dtype=False)
+
+
+def test_inverse_with_X_negative_and_positive():
+    X = pd.DataFrame(
+        {
+            "var1": np.arange(-20, 0),
+            "var2": np.arange(0, 20),
+            "var3": np.arange(-10, 10),
+        }
+    )
+
+    transformer = YeoJohnsonTransformer(variables=None)
+    X_trans = transformer.fit_transform(X)
+
+    X_inverse = transformer.inverse_transform(X_trans)
+
+    pd.testing.assert_frame_equal(X, X_inverse, check_dtype=False)
+
+
+def test_lambda_equals_lambda_equal_0():
+    X = pd.DataFrame(
+        {
+            "var1": np.arange(0, 20),
+            "var2": np.arange(20, 40),
+        }
+    )
+
+    transformer = YeoJohnsonTransformer(variables=None)
+    transformer = transformer.fit(X)
+
+    transformer.lambda_dict_ = {"var1": 0, "var2": 0}
+
+    X_trans = transformer.transform(X)
+    X_inverse = transformer.inverse_transform(X_trans)
+
+    pd.testing.assert_frame_equal(X, X_inverse, check_dtype=False)
```

Comment:
Hey @GiorgioSgl 

Thank you so much for the revised version of the code. We are almost there!!

I am not sure I understand this test. Maybe you can help me.

in line 126, the transformer is fit to X, hence it learns some lambdas. Which are the lambda values? Not 0, are they?

in line 128, we re-define the lambda dictionary, because we want to test the inverse_transform functionality for lambda 0, in short, that when lambda is zero, this tests that inverse transform applies the correct calculation.

So my idea here was, to apply inverse_transform to X, it will return some values. Then, we need to perform the calculations manually, to know what the transformation of X would return when it enters the inverse transform with lambda zero, and then assert that X_inverse is the same as a pandas dataframe, that we would hard-code with the expected values.

But here, something else is happening. What was your idea?

The same for the coming test.

Thank you for your help!

Reply:
Not sure what I was thinking here, but the tests are great. Thank you!